src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -67,6 +67,25 @@ public static ValueTask<int> ReadAsync(this IPipeReader input, Span<byte> destin
             return new ValueTask<int>(input.ReadAsyncAwaited(destination));
         }
 
+        public static Task CopyToEndAsync(this IPipeReader input, Stream stream)
+        {
+            return input.CopyToEndAsync(stream, 4096, CancellationToken.None);
+        }
+
+        public static async Task CopyToEndAsync(this IPipeReader input, Stream stream, int bufferSize, CancellationToken cancellationToken)
+        {
+            try
+            {
+                await input.CopyToAsync(stream, bufferSize, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                input.Complete(ex);
+                return;
+            }
+            return;
```

Comment:
Awesome. I was also wondering if AsPipelineWriter should be calling this method instead of CopyToAsync() so exceptions are handled.